Branch-SDK/BNCSKAdNetwork.m

```diff
@@ -0,0 +1,80 @@
+//
+//  BNCSKAdNetwork.m
+//  Branch
+//
+//  Created by Ernest Cho on 8/12/20.
+//  Copyright © 2020 Branch, Inc. All rights reserved.
+//
+
+#import "BNCSKAdNetwork.h"
+#import "BNCApplication.h"
+
+@interface BNCSKAdNetwork()
+
+@property (nonatomic, copy, readwrite) NSDate *installDate;
+
+@property (nonatomic, strong, readwrite) Class skAdNetworkClass;
+@property (nonatomic, assign, readwrite) SEL skAdNetworkRegisterAppForAdNetworkAttribution;
+@property (nonatomic, assign, readwrite) SEL skAdNetworkUpdateConversionValue;
+
+@end
+
+@implementation BNCSKAdNetwork
+
++ (BNCSKAdNetwork *)sharedInstance {
+    static BNCSKAdNetwork *singleton;
```

Comment:
Should this use `@synchronized(self)` around the whole thing? I think it is theoretically possible for two racing threads to try to initialize the `static` vars simultaneously.